File: CsFormsFlashcards/CsFormsFlashcards/FlashcardsCls.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.IO;

namespace CsFormsFlashcards
{
    interface IImagen
    {
        string CargarImagen();
    }

    abstract class FlashcardsCls
    {
        public abstract string[] obtenerdatos();

        public string mostrarpregunta()
        {
            string prg = obtenerdatos()[0];

            return prg;
        }

        public string mostrarrespeusta()
        {
            string resp = obtenerdatos()[1];
            return resp;
        }


    }

    class texto : FlashcardsCls
    {
        StreamReader Archivo;
        public override string[] obtenerdatos()
        {
            string[] filess = Directory.GetFiles("C:/Users/oscar/Desktop/Puntaje/");
            Random rdm = new Random();
            int pp = rdm.Next(0, filess.Length);

            string spregut = filess[pp];

             Archivo = new StreamReader(filess[pp]);
            string respuesta = Archivo.ReadToEnd();

            string[] pregResp = new string[2] { spregut.Replace("C:/Users/oscar/Desktop/Puntaje/", "").Replace(".txt", ""), respuesta };

            return pregResp;
        }

        ~texto()
        {
            Archivo.Close();
        }
    }
   

    class Imagen : FlashcardsCls, IImagen
    {
        public string CargarImagen()
        {
            string Soloruta = obtenerdatos()[1];

            return Soloruta;
        }

        public override string[] obtenerdatos()
        {

            string[] imajes = Directory.GetFiles("C:/Users/oscar/Desktop/Imagen");

            texto tn = new texto();
            string rut = "C:/Users/oscar/Desktop/Imagen" + "/" + tn.mostrarpregunta() + ".jpg";
            
            string pap =  Convert.ToString(imajes.Contains(rut));

            string[] cont = new string[2] { pap, rut };
                    
            
            return cont;
        }

    }
}

File: VbFormsFlashcards/VbFormsFlashcards/FlashcardsCls.vb
Imports System.IO

Interface IImagen

    Function CargarImagen() As String

End Interface
Public MustInherit Class FlashcardsCls

    Public MustOverride Function obtenerdatos() As String()

    Public Function mostrarpregunta() As String

        Dim prg As String = obtenerdatos()(0)

        Return prg

    End Function

    Public Function mostrarrespeusta()

        Dim resp = obtenerdatos()(1)

        Return resp

    End Function

End Class

Public Class texto
    Inherits FlashcardsCls

    Dim Archivo As StreamReader

    Public Overrides Function obtenerdatos() As String()

        Dim filess() As String = Directory.GetFiles("C:/Users/oscar/Desktop/Puntaje/")

        Dim rdm As New Random
        Dim pp As Integer = rdm.Next(0, filess.Length)

        Dim spregut As String = filess(pp)

        Archivo = My.Computer.FileSystem.OpenTextFileReader(filess(pp))
        Dim respuesta = Archivo.ReadToEnd()

        Dim pregResp() As String = {spregut.Replace("C:/Users/oscar/Desktop/Puntaje/", "").Replace(".txt", ""), respuesta}

        Return pregResp

    End Function
End Class

Public Class Imagen
    Inherits FlashcardsCls
    Implements IImagen

    Public Overrides Function obtenerdatos() As String()

        Dim imajes() As String = Directory.GetFiles("C:/Users/oscar/Desktop/Imagen")

        Dim tn As New texto()

        Dim rut As String = "C:/Users/oscar/Desktop/Imagen" & "/" & tn.mostrarpregunta() & ".jpg"

        Dim pap As String = Convert.ToString(imajes.Contains(rut))

        Dim cont() As String = {pap, rut}

        Return cont

    End Function

    Public Function CargarImagen() As String Implements IImagen.CargarImagen

        Dim Soloruta As String = obtenerdatos()(1)

        Return Soloruta

    End Function
End Class



